### PR TITLE
Dependency cleanup

### PR DESCRIPTION
**Pull request by johnmay**
First round of dependency cleanup. Tediously using dependency:analyze to make sure everything used is declared and everything unused is not declared. Cleans up tool/ and descriptor/ modules a fair bit.

### CODE REVIEW

**Comment by johnmay** [comment]
@egonw Can you check if it works locally for you? not sure why the action is failing

**Comment by egonw** [comment]
Compiling works here. Busy with kids...

**Comment by johnmay** [comment]
No problem will fire up old laptop and check later

**Comment by johnmay** [comment]
I think an XML lib is now part of JDK 17... curious though the iordf did not explicitly include it so was bringing it in from elsewhere

**Comment by johnmay** [comment]
Aight we good now :-)

File: misc/log4j/src/main/java/org/openscience/cdk/tools/LoggingTool.java [hunk 1]
```diff
@@ -170,13 +193,20 @@ public LoggingTool(Class<?> classInst) {
...
+                            LoggerContext.getContext(true),
+                            new ConfigurationSource(resourceAsStream));
+                    Configurator.reconfigure(config);
+                }
+            }
         } catch (NullPointerException e) { // NOPMD
             localLogger.error("Properties file not found: ", e.getMessage());
             localLogger.debug(e);
         } catch (Exception e) {
+            e.printStackTrace();
```
Comment (egonw) [inline misc/log4j/src/main/java/org/openscience/cdk/tools/LoggingTool.java:209]: Is this intended?

Comment (egonw) [reply misc/log4j/src/main/java/org/openscience/cdk/tools/LoggingTool.java:209]: (can be fixed later if not)

**Review by egonw — APPROVED** [review]

**Comment by egonw** [comment]
@johnmay, sorry for the delay.

File: misc/log4j/src/main/java/org/openscience/cdk/tools/LoggingTool.java (hunk 1 above)
Comment (johnmay) [reply misc/log4j/src/main/java/org/openscience/cdk/tools/LoggingTool.java:209]: Good spot - was fighting log4j2 config loading. The doc is very bad

**Comment by johnmay** [comment]
No delay or rush at all. Will likely have a fair few more coming but don’t rush to get them merged just putting them up as good time to do the cleanup for me

**Pull request merged**